### ISSUE TITLE
queue job No query results for model

### ISSUE THREAD

**Issue by juchao1989** [laravel/framework#24409]
- Laravel Version: 5.5
- PHP Version:7.0
- Database Driver & Version:mysql

### Description: !!!please help me!!!
!!!!Frequency of occurrence, every 1000 times!!!! 
 !!!on sqs queue!!!
!!!on redis queue!!!

Illuminate\Database\Eloquent\ModelNotFoundException: No query results for model [App\Models\Transaction]. in /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:416
Stack trace:
#0 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/SerializesAndRestoresModelIdentifiers.php(54): Illuminate\Database\Eloquent\Builder->firstOrFail()
#1 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/SerializesModels.php(41): App\Jobs\DealingWithTransactions->getRestoredPropertyValue(Object(Illuminate\Contracts\Database\ModelIdentifier))
#2 [internal function]: App\Jobs\DealingWithTransactions->__wakeup()
#3 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(42): unserialize('O:32:"App\\Jobs\\...')
#4 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(76): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\SqsJob), Array)
#5 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(320): Illuminate\Queue\Jobs\Job->fire()
#6 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(270): Illuminate\Queue\Worker->process('usdt_btc', Object(Illuminate\Queue\Jobs\SqsJob), Object(Illuminate\Queue\WorkerOptions))
#7 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(114): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\SqsJob), 'usdt_btc', Object(Illuminate\Queue\WorkerOptions))
#8 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon('usdt_btc', 'BtcUsdtMatcheng...', Object(Illuminate\Queue\WorkerOptions))
#9 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\Queue\Console\WorkCommand->runWorker('usdt_btc', 'BtcUsdtMatcheng...')
#10 [internal function]: Illuminate\Queue\Console\WorkCommand->handle()
#11 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#12 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#14 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#15 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call(Array)
#16 /var/www/api_imex/vendor/symfony/console/Command/Command.php(252): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#17 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#18 /var/www/api_imex/vendor/symfony/console/Application.php(946): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/api_imex/vendor/symfony/console/Application.php(248): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/api_imex/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /var/www/api_imex/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 /var/www/api_imex/artisan(37): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#24 {main}

**Comment by sisve** [comment]
This happens because you have a job that works with a Transaction model, and that model doesn't exist in the database when the job executes. If you think this is an issue, provide us with working code under the header "Steps To Reproduce". You're not supposed to leave that one blank.

**Comment by juchao1989** [comment]
@sisve I am sorry, my english is poor!!!
I wrote a job, put it in the queue, performed about 1000 times, there will be a case where the model could not be found, not every time...    Thank you for your reply

**Comment by wuqinqiang** [comment]
I have the same problem,The problem is difficult to reproduce,but it is real.

> Laravel Version: 5.5

Just a quick look at the code：
```
\DB::transaction(function () use ($request, $user, $ids) {
            collect($ids)->map(function ($id) use ($user, $request) {
                //...some code 
                //....some code 
             $data=['amount'=>100,'user_id'=>211];
             $subPayment = SubPayment::create(data);
             SendMqPayOrRefund::dispatch($sub_payment, 1)
               //...some code 
            });
        });
```
Sometimes (low frequency)

`
/www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/SerializesAndRestoresModelIdentifiers.php(54): Illuminate\Database\Eloquent\Builder->firstOrFail()
#1 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/SerializesModels.php(41): App\Jobs\SendMqPayOrRefund->getRestoredPropertyValue(Object(Illuminate\Contracts\Database\ModelIdentifier))
#2 [internal function]: App\Jobs\SendMqPayOrRefund->__wakeup()
#3 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(42): unserialize('O:26:"App\\Jobs\\...')
#4 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(76): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\RedisJob), Array)
#5 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(320): Illuminate\Queue\Jobs\Job->fire()
#6 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(270): Illuminate\Queue\Worker->process('redis', Object(Illuminate\Queue\Jobs\RedisJob), Object(Illuminate\Queue\WorkerOptions))
#7 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(114): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\RedisJob), 'redis', Object(Illuminate\Queue\WorkerOptions))
#8 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon('redis', 'default', Object(Illuminate\Queue\WorkerOptions))
#9 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\Queue\Console\WorkCommand->runWorker('redis', 'default')
#10 [internal function]: Illuminate\Queue\Console\WorkCommand->handle()
#11 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#12 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#14 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#15 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call(Array)
#16 /www/wwwroot/zhongxin/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#17 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#18 /www/wwwroot/zhongxin/vendor/symfony/console/Application.php(969): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /www/wwwroot/zhongxin/vendor/symfony/console/Application.php(255): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /www/wwwroot/zhongxin/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 /www/wwwroot/zhongxin/artisan(37): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#24 {main}, 2020-03-15 21:57:25)) {"exception":"[object] (Illuminate\\Database\\QueryException(code: HY000): SQLSTATE[HY000]: General error: 1064 [10520fd65c0fa001][10.0.14.131:3306][hstsaas]ERR-CODE: [TDDL-4518][ERR_VALIDATE] ERR-CODE: [TDDL-4007][ERR_CANNOT_FETCH_TABLE_META] Table 'failed_jobs' metadata cannot be fetched because com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'hstsaas_is2g_0000.failed_jobs' doesn't exist. .  (SQL: insert into `failed_jobs` (`connection`, `queue`, `payload`, `exception`, `failed_at`) values (redis, default, {\"displayName\":\"App\\\\Jobs\\\\SendMqPayOrRefund\",\"job\":\"Illuminate\\\\Queue\\\\CallQueuedHandler@call\",\"maxTries\":null,\"timeout\":null,\"timeoutAt\":null,\"data\":{\"commandName\":\"App\\\\Jobs\\\\SendMqPayOrRefund\",\"command\":\"O:26:\\\"App\\\\Jobs\\\\SendMqPayOrRefund\\\":9:{s:8:\\\"\\u0000*\\u0000order\\\";O:45:\\\"Illuminate\\\\Contracts\\\\Database\\\\ModelIdentifier\\\":3:{s:5:\\\"class\\\";s:21:\\\"App\\\\Models\\\\SubPayment\\\";s:2:\\\"id\\\";i:11881;s:10:\\\"connection\\\";s:5:\\\"mysql\\\";}s:13:\\\"\\u0000*\\u0000is_request\\\";i:1;s:6:\\\"\\u0000*\\u0000job\\\";N;s:10:\\\"connection\\\";N;s:5:\\\"queue\\\";N;s:15:\\\"chainConnection\\\";N;s:10:\\\"chainQueue\\\";N;s:5:\\\"delay\\\";N;s:7:\\\"chained\\\";a:0:{}}\"},\"id\":\"3932\",\"attempts\":0,\"type\":\"job\",\"tags\":[\"App\\\\Models\\\\SubPayment:11881\"],\"pushedAt\":1584280645.785931}, Illuminate\\Database\\Eloquent\\ModelNotFoundException: **No query results for model [App\\Models\\SubPayment]**. in /www/wwwroot/zhongxin/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:414


`

the Job

```
<?php

namespace App\Jobs;

use App\Models\SubPayment;
use App\Service\RocketMq;
use Illuminate\Bus\Queueable;
use Illuminate\Queue\SerializesModels;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Support\Facades\Log;

class SendMqPayOrRefund implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    protected $order;

    protected $is_request;

    /**
     *
     * @return void
     */
    public function __construct($order, $is_request)
    {
        $this->order = $order;
        $this->is_request = $is_request;
    }

    /**
     *    * @return void
     */
    public function handle()
    {
        //....some code
    }
}

```

Please tell me how to solve it. Thank you

**Comment by wuqinqiang** [comment]
@sisve

**Comment by netfireapo** [comment]
@wuqinqiang 

because you run multiple times the job add some seconds `delay`

in your Job

```
public function __construct($media)
    {
        $this->media = $media;
        $this->onQueue('media');
        $this->delay(5);
    }
```

**Comment by puzzledmonkey** [comment]
hi and sorry for the necro but i've also found this issue, but only when setting 'receive message wait time' to anything above 0 in the SQS settings. the lower the wait time the less the chances of it happening are (at 20 seconds it seems to happen quite a bit more often but it's hard to quantify), but when the wait time is 0 it never seems to occur.

App\Notifications\BankTransferOrderReceived failed with error No query results for model [App\Models\Order].

we NEVER delete any Orders from the system so there is no chance that it's a race condition with some deletion. Orders are definitely saved before the BankTransferOrderReceived is triggered. However, Orders are saved within a transaction block, I'm not sure if this has any effect on the issue?

So to recap, with a SQS queue with Receive Message Wait Time >0, using Redis queue and a transaction to save the models, there is a rare chance of getting he 'No query results' when trying to process the job. Since it's apparently random I suspect a race condition between the cache and queue systems but I have no real idea where to start investigating in the Laravel code.

Thanks for any suggestions!
